src/SignalR/common/Http.Connections/src/Internal/Transports/LongPollingServerTransport.cs

```diff
@@ -35,6 +39,8 @@ public LongPollingServerTransport(CancellationToken timeoutToken, PipeReader app
 
     public async Task ProcessRequestAsync(HttpContext context, CancellationToken token)
     {
+        AddNoCacheHeaders(context.Response);
```

Comment:
I think we might also need to do this for the first poll response here.

https://github.com/dotnet/aspnetcore/blob/a450cb69b5e4549f5515cdb057a68771f56cefd7/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionContext.cs#L430-L444

Are there any other requests we need to disable caching for? I think `/negotiate` is fine as-is because all the clients POST with an empty request body AFAIK.

Reply:
> I think we might also need to do this for the first poll response here.

It turns out that the code for setting headers to prevent caching should be put in a separate utility class. Should it be [src/Http/Headers/src/HeaderUtilities.cs](https://github.com/dotnet/aspnetcore/blob/73db5e40bbdeb4f04e5e05b9090f253760d1e1ee/src/Http/Headers/src/HeaderUtilities.cs)?

Reply:
Oh, I saw that all requests are handled in one place:

https://github.com/dotnet/aspnetcore/blob/a450cb69b5e4549f5515cdb057a68771f56cefd7/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs#L79-L98

At here we can add headers to prevent caching for all requests, including first polls, as well as for Disconnect requests.

https://github.com/dotnet/aspnetcore/blob/a450cb69b5e4549f5515cdb057a68771f56cefd7/src/SignalR/common/Http.Connections/src/Internal/HttpConnectionDispatcher.cs#L506

Reply:
I don't think we need to add headers to POST or DELETE requests because those are not cacheable to begin with unless there are Cache-Control/Expires response headers stating otherwise.

>    If the request passes through a cache and the Request-URI identifies
>   one or more currently cached entities, those entries SHOULD be
>   treated as stale. Responses to this method are not cacheable.

https://datatracker.ietf.org/doc/html/rfc2616#section-9.7

Reply:
In addition to the first poll request, I found that GET requests can get 4xx responses in the following cases:
- 400, when no ConnectionId provided
- 404, when unknown ConnectionId provided
- 404, when LongPolling transport isn't supported

I also added cache headers for them and extended the tests.

@halter73, thank you for making me look into it.